walletnode/services/artworksearch/request.go

```diff
@@ -0,0 +1,63 @@
+package artworksearch
+
+// ArtSearchQueryField represents artsearch query-able field
+type ArtSearchQueryField string
+
+// List of types of activation ticket.
+const (
+	ArtSearchArtistName ArtSearchQueryField = "artist_name"
+	ArtSearchArtTitle   ArtSearchQueryField = "art_title"
+	ArtSearchSeries     ArtSearchQueryField = "series"
+	ArtSearchDescr      ArtSearchQueryField = "descr"
+	ArtSearchKeyword    ArtSearchQueryField = "keyword"
+)
+
+// ArtSearchQueryFields returns all query-able field names for an artwork
+var ArtSearchQueryFields = []string{
+	ArtSearchArtistName.String(),
+	ArtSearchArtTitle.String(),
+	ArtSearchSeries.String(),
+	ArtSearchDescr.String(),
+	ArtSearchKeyword.String(),
+}
+
+// String returns the string val of ArtSearchQueryField
+func (f ArtSearchQueryField) String() string {
+	return string(f)
+}
+
+// ArtSearchRequest represents artwork search payload
+type ArtSearchRequest struct {
```

Comment:
i am just curious why this structure uses many pointer fields? it's necessary?

Reply:
Yes. This is incoming request payload. the fields that correspond to parameters that are optional are defined as pointers. Because if a `param` is provided & we don't use pointers then that `param` will come in as `false` which will make it impossible to identify the true intention of user.